src/hooks/useTimer.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState, useRef, useCallback, useEffect } from 'react';
+
+export type TimerType = 'countdown' | 'tabata' | 'stopwatch' | 'xy';
+
+export type TimerSequenceItem = {
+    type: TimerType;
+    initialTime?: number;
+    rounds?: number;
+    workDuration?: number;
+    breakDuration?: number;
+    roundMinutes?: number;
+    roundSeconds?: number;
+    label?: string;
+};
+
+const useTimer = () => {
```

Comment:
I like the intention of creating a generic custom hook for managing timers. However, this hook encapsulates state locally but does not make it available from anywhere other than where the hook is called. This means you then have to pass props down to child components, which is not generally a good practice (remember trying to avoid prop drilling https://react.dev/learn/passing-data-deeply-with-context). If you used context instead, all children would have access to these methods.